docker-compose.yml

```diff
@@ -0,0 +1,15 @@
+version: '2'
+services:
+  calypso:
+    image: node:6.9.1
+    environment:
+      CALYPSO_ENV: 'development'
+    ports:
+      - '3000:3000'
+    volumes:
+      - ./:/calypso
+      - node_modules:/calypso/node_modules
```

Comment:
bind mount volumes that you're mostly familiar with take a directory on the host and bind mount in the running container. A named volume, is a volume that remains between invocations and can be shared across running containers (similar to a bind volume). The difference is that it's not directly accessible by the host. You can configure a named volume to use AWS EBS, AWS S3, or any other block storage that there is a plugin for.

In this case, we aren't using a plugin, we are simply saying "create and manage a volume named 'node_modules'".

The reason I'm choosing to use a named volume, instead of a bind mount has to do with npm and filesystem boundaries/types. I'm running in windows, so I can't bind mount my node_modules directory directly since the linux container can't run `.exe` files. Even if I bind mounted the node_modules directory elsewhere, I would run into issues with filenames being case sensitive and some other weird things I've run into in the past.

With a named volume, the node_modules directory effectively is guaranteed to be a real linux filesystem no matter what OS I'm using. So, it offers a guarantee of less issues being filed in the future. I should probably put this in some documentation :D